### DISCUSSION TITLE
imgui demo (d3d9) crashes when pressing keyboard button

### DISCUSSION THREAD

**Discussion by ghost** [ocornut/imgui#5879] [New Users Build/Link/Run issues ONLY!]
Hello together,

I copy paste the dx9 example and everything works/looks fine. Except when I press any keyboard button, then I get an error:
![grafik](https://user-images.githubusercontent.com/44911038/201038449-5dd6c6ab-3adb-47a2-85b0-72f06f1ce9f1.png)

Debug Console output:
Assertion failed: ImGui::IsNamedKey(key), file _(path)_\vcpkg\buildtrees\imgui\src\v1.88-34c93572a3.clean\imgui.cpp, line 1280

Can somebody help me pls?

My imgui version 1.88

**Comment by ocornut** [comment]
Hello,
I cannot seem to repro the issue here.

- Could you open Demo->Tools->About and copy and paste the blurb here?
- What is the value of `key` in the debugger?
- How are you building "Dear ImGui DirectX9 Example" if your copy of imgui comes from vcpkg? I don't assume vcpkg include the examples. If you are building "Dear ImGui DirectX9 Example" from our repo presumably it should use the version in our repo and not from vcpkg?

**Comment by ghost** [comment]
Hello ocornut,

thanks for your fast reply.
Your third point reminded me that with the installation with vcpkg I only had the imgui.h and manually downloaded the imgui_impl_dx9 and imgui_impl.win32 and added them to my project. This was not the best idea, since apparently the two manually downloaded parts caused the problem.

**For the solution** I removed the old imgui_impl_dx9/imgui_impl.win32 files and replaced them with the fresh downloaded realese 1.88. Now the error is fixed.

I'm not using vcpkg for so long, is there a reason why I don't install the backend data with the imgui installation with vcpkg?


If you still want to see the about blurb:

```
Dear ImGui 1.88 (18800)
--------------------------------
sizeof(size_t): 4, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _MSC_VER=1932
define: _MSVC_LANG=201402
--------------------------------
io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx9
io.ConfigFlags: 0x00000000
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000000E
 HasMouseCursors
 HasSetMousePos
 RendererHasVtxOffset
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1264.00,761.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

Thank you very much for the help :)
